### PR TITLE
Add rbd-mirror health status

### PR DESCRIPTION
**Pull request by concubidated**

### CODE REVIEW

File: ceph/rbd_mirror_status.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+)
+
+const rbdPath = "/usr/bin/rbd"
+
+const (
+	// RbdMirrorOK denotes the status of the rbd-mirror when healthy.
+	RbdMirrorOK = "OK"
```
Comment (Matt1360) [inline ceph/rbd_mirror_status.go:29]: Just a general question, do these strings change between Nautilus and Pacific?  Same question about the struct layout below.

File: ceph/rbd_mirror_status.go [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+// Collect sends all the collected metrics Prometheus.
+func (c *RbdMirrorStatusCollector) Collect(ch chan<- prometheus.Metric) {
+	status, err := rbdMirrorStatus(c.config)
+	if err != nil {
+		c.logger.WithError(err).Error("failed to run 'rbd mirror pool status'")
+	}
+	var rbdStatus rbdMirrorPoolStatus
+	if err = json.Unmarshal(status, &rbdStatus); err != nil {
+		//blabla
```
Comment (Matt1360) [inline ceph/rbd_mirror_status.go:142]: https://www.youtube.com/watch?v=Hrph2EW9VjY

Comment (concubidated) [reply ceph/rbd_mirror_status.go:142]: oops

File: ceph/rbd_mirror_status.go (hunk 1 above)
Comment (concubidated) [reply ceph/rbd_mirror_status.go:29]: the struct actually doesn't have daemon and image health separate, so I need to double check/test that on nautilus.

Comment (concubidated) [reply ceph/rbd_mirror_status.go:29]: yes, still works fine on nautilus

File: ceph/rbd_mirror_status.go (hunk 2 above)
Comment (baergj) [reply ceph/rbd_mirror_status.go]: I have that album!

**Review by Matt1360 — APPROVED** [review]

File: ceph/exporter.go [hunk 3]
```diff
@@ -86,6 +91,42 @@ func (exporter *Exporter) cephVersionCmd() []byte {
...
+func (exporter *Exporter) cephVersionsCmd() []byte {
+	cmd, err := json.Marshal(map[string]interface{}{
+		"prefix": "versions",
+	})
+	if err != nil {
+		exporter.Logger.WithError(err).Panic("failed to marshal ceph versions command")
+	}
+
+	return cmd
+}
```
Comment (baergj) [inline ceph/exporter.go:103]: This is also implemented in `monitor.go`, including parsing. Maybe factor it out into a common place? Ideally we would run each command exactly once but I think that's a bit more work, so it's fine if it runs twice.

Comment (concubidated) [reply ceph/exporter.go]: Okay, I think I got this a little better. i moved the cmd call to exporter and made it public, then added the parse function there as well.

File: ceph/exporter.go [hunk 4]
```diff
@@ -86,6 +93,84 @@ func (exporter *Exporter) cephVersionCmd() []byte {
...
+		return err
+	}
+
+	// check to see if rbd-mirror is in ceph version output and not empty
+	if _, exists := versions["rbd-mirror"]; exists {
+		version := versions["rbd-mirror"]
+		if len(version) != 0 {
+			exporter.RbdMirror = true
+		}
+	}
```
Comment (baergj) [inline ceph/exporter.go:169]: Because the length of a `nil` map is `0`, you could shorten this to:
```go
if len(versions["rbd-mirror"]) > 0 {
    exporter.RbdMirror = true
}
```
if you wanted.

**Review by baergj — APPROVED** [review]

File: ceph/exporter.go (hunk 3 above)
Comment (baergj) [reply ceph/exporter.go]: :+1: If you wanted to take it one step further you could roll the run-command-and-parse into one function, but that's up to you.

File: ceph/exporter.go (hunk 4 above)
Comment (concubidated) [reply ceph/exporter.go:169]: I shall do that!
